session/isolation_test.go

```diff
@@ -226,6 +475,44 @@ func (s *testIsolationSuite) TestA5BWriteSkew(c *C) {
 	session2.MustExec("update x set c = c+1 where id = 1;")
 	session2.MustExec("commit;")
 	session1.MustExec("commit;")
+
+	session1.MustExec("set tidb_txn_mode = 'pessimistic'")
+	session2.MustExec("set tidb_txn_mode = 'pessimistic'")
+
+	session1.MustExec("drop table if exists x;")
+	session1.MustExec("drop table if exists y;")
+	session1.MustExec("create table x (id int primary key, c int);")
+	session1.MustExec("insert into x values(1, 1);")
+	session1.MustExec("create table y (id int primary key, c int);")
+	session1.MustExec("insert into y values(1, 1);")
+
+	session1.MustExec("begin;")
+	session2.MustExec("begin;")
+	session1.MustQuery("select c from x where id = 1;").Check(testkit.Rows("1"))
+	session2.MustQuery("select c from y where id = 1;").Check(testkit.Rows("1"))
+	session1.MustExec("update y set c = c+1 where id = 1;")
+	session2.MustExec("update x set c = c+1 where id = 1;")
+	session2.MustExec("commit;")
+	session1.MustExec("commit;")
+
+	session1.MustExec("set tx_isolation = 'READ-COMMITTED'")
+	session2.MustExec("set tx_isolation = 'READ-COMMITTED'")
+
+	session1.MustExec("drop table if exists x;")
+	session1.MustExec("drop table if exists y;")
+	session1.MustExec("create table x (id int primary key, c int);")
+	session1.MustExec("insert into x values(1, 1);")
+	session1.MustExec("create table y (id int primary key, c int);")
+	session1.MustExec("insert into y values(1, 1);")
+
+	session1.MustExec("begin;")
+	session2.MustExec("begin;")
+	session1.MustQuery("select c from x where id = 1;").Check(testkit.Rows("1"))
+	session2.MustQuery("select c from y where id = 1;").Check(testkit.Rows("1"))
+	session1.MustExec("update y set c = c+1 where id = 1;")
```

Comment:
Should it be like this?
```
x.c1 <- 1  y.c1 <- 2
session1                                    session2
begin
                                                   begin
x.c1 <- y.c1                          
                                             y.c1 <- x.c1
commit
                                                    commit
```
and verify  `x.c1` and `y.c1` get swapped but not the same

Reply:
done